nntrainer/src/tensor.cpp

```diff
@@ -66,7 +66,17 @@ Tensor::Tensor(const TensorDim &d, float *buf) :
   dim(d),
   strides{{1, 2, 3}},
   is_contiguous(true),
-  data(new float[d.getDataLen()], std::default_delete<float[]>()) {
+  data(std::shared_ptr<float>(static_cast<float *>(nullptr))) {
```

Comment:
```suggestion
  data(nullptr) {
```

Reply:
Or how about this to avoid reinitialization

```suggestion
  data(d.getDataLen() == 0 ? nullptr: std::shared_ptr<float> ...  )) {
```

Reply:
Done.